### ISSUE TITLE
Newline etc, escaping bug.

### ISSUE THREAD

**Issue by dwarring** [timo/json_fast#35]
% perl6 -v
    This is Rakudo version 2017.12-88-g8fd776fd7 built on MoarVM version 2017.12-15-g8414888b implementing Perl 6.c.
    % perl6 -I ../json_fast/lib/ -MJSON::Fast -e'say from-json(q<"AB\nC">).perl'
    "AB\nC"
    % perl6 -I ../json_fast/lib/ -MJSON::Fast -e'say from-json(q<"AB\nC\u000A">).perl'
    "ABnC\n"

Note the decoding of `\n` as `n`, but only if '\n`and `\u000A` are used together in a string.

New regression in JSON::Fast 0.9.9.

**Comment by mienaikage** [comment]
It's looking like any use of `\u` and `\n` together in the same string is breaking newlines in `0.9.9`
```
$ perl6 -e 'use JSON::Fast:ver<0.9.8>; dd from-json(‘{"test":"\n backslash u: \u2605 \n","test2":"\n no backslash u: ★ \n"}’)'
${:test("\n backslash u: ★ \n"), :test2("\n no backslash u: ★ \n")}

$ perl6 -e 'use JSON::Fast:ver<0.9.9>; dd from-json(‘{"test":"\n backslash u: \u2605 \n","test2":"\n no backslash u: ★ \n"}’)'
${:test("n backslash u: ★ n"), :test2("\n no backslash u: ★ \n")}
```

**Comment by zoffixznet** [comment]
Possibly related to https://github.com/timo/json_fast/issues/36

**Comment by timo** [comment]
i believe i fixed this in release 0.9.10; please try it on your real-life datasets

**Comment by dwarring** [comment]
Now working. Thanks.
